### PR TITLE
162 - store and check expires_in time for the access_token  

### PR DESCRIPTION
**Pull request by jonathanchrisp**
## Changes

* Added new method to check if the access_token has been expired and added a check within the authenticated? method. 
* Added new instance variable on the client instance (access_token_expiry) which returns the date

## Results
Ran the unit and rubocop tests as per the contriibuting guidelines:
```
108 files inspected, no offenses detected
Finished in 0.02969 seconds (files took 0.99518 seconds to load)
41 examples, 0 failures
```

### CODE REVIEW

**Comment by jonathanchrisp** [comment]
@benSlaughter - initial work on this as workload was light today. Let me know what you think, please don't merge as of yet as I think some basic unit tests should be added for the new methods.

**Comment by jonathanchrisp** [comment]
@benSlaughter - I've beefed up :cow2: the unit tests so happy for this to be reviewed.

**Comment by jonathanchrisp** [comment]
@benSlaughter - let me know if you can review this. I would like to try and get a newer version of the gem out as we have done so many updates. If possible can you or @bw-lee let me know how many endpoints you still have to add?

File: lib/bwapi/client.rb [hunk 1]
```diff
@@ -65,11 +65,23 @@ def initialize(options = {})
       end
     end
 
+    # Check is access token has expired
+    #
+    # @return [Boolean] access token expiry status
+    def access_token_expired?
+      return true if @access_token_expiry.nil?
+      seconds_until_access_token_expires > 0 ? false : true
```
Comment (benSlaughter) [inline lib/bwapi/client.rb]: does this need to be a ternary operator, as you could just use:
`seconds_until_access_token_expires <= 0`

**Comment by benSlaughter** [comment]
One minor comment, other than that looks great :+1:

**Comment by jonathanchrisp** [comment]
@benSlaughter - updated - please review again :tada:

**Pull request merged**